### PR TITLE
Exception handling corrections

### PR DESCRIPTION
**Pull request by ArthaTi**
Botto will no longer crash if something goes wrong in the API, for example it posts to the wrong channel or crosses 2000 characters.

### CODE REVIEW

File: src/discord/cache.js [hunk 1]
```diff
@@ -60,8 +60,9 @@ class DiscordCache {
 
   #load () {
     const rawFile = fs.readFileSync(cacheFilePath)
-    const messages = JSON.parse(rawFile).messages
-    const channels = JSON.parse(rawFile).caches
+    const json = JSON.parse(rawFile);
```
Comment (williammck) [inline src/discord/cache.js:63]:
```suggestion
    const json = JSON.parse(rawFile)
```

bad semicolon, nobody likes you

File: src/discord/cache.js [hunk 2]
```diff
@@ -60,8 +60,9 @@ class DiscordCache {
...
+    const messages = json.messages
+    const channels = json.channels
```
Comment (williammck) [inline src/discord/cache.js:65]: can we have it pull from `caches` if that's present? would make the current file compatible

File: src/index.js [hunk 3]
```diff
@@ -30,267 +30,277 @@ const sendMessage = async function (opts) {
 }
 
 bot.on('messageCreate', async function (msginst) {
-  let message = msginst.content
-  const userID = msginst.author.id
-  const channelID = msginst.channelId
+  try {
```
Comment (williammck) [inline src/index.js:33]: idk how I feel about wrapping this whole function in a try/catch. wonder if there's a nicer way to do this... i think you can do `bot.on('error', function (err) { ... })` and any uncaught exception from the EventEmitter will end up in there

side note, i'm looking at using https://www.sapphirejs.dev/ for commands and such, haven't investigated its error handling yet.

File: src/index.js [hunk 4]
```diff
@@ -30,267 +30,277 @@ const sendMessage = async function (opts) {
...
+    }
 
-  const reply = dcache.getMessageFromChannel(channelID)
-  if (channels.isChatChannel(channelID) || Math.random() < chatProbability || checks.hasMention(message)) {
-    markovInProgress = true
+    const reply = dcache.getMessageFromChannel(channelID)
+    if (channels.isChatChannel(channelID) || Math.random() < chatProbability || checks.hasMention(message)) {
 
-    await new Promise(r => setTimeout(r, Math.round(Math.random() * 10000)))
+      try {
```
Comment (williammck) [inline src/index.js:270]: ditto my above comment

**Review by williammck — CHANGES_REQUESTED** [review]

**Comment by ArthaTi** [comment]
oops i must have introduced some local changes that weren't supposed to be on this branch

**Comment by ArthaTi** [comment]
I've completely lost interest in running botto and fixing this

**Pull request closed without merging**